### PR TITLE
Search python fixers in virtual environment

### PR DESCRIPTION
**Pull request by jlesquembre**
For the python linters, ALE will search first in a virtualenv first. This commit extends the python fixers to do the same, first try to find the fixer command in a virtualenv.
If the command is not found (nor in the virtualenv nor globally), don't do anything. Now if the fixer command doesn't exist, `ALEFix` returns an empty file.

### CODE REVIEW

File: autoload/ale/handlers/python.vim [hunk 1]
```diff
@@ -45,3 +45,24 @@ function! ale#handlers#python#HandlePEP8Format(buffer, lines) abort
...
+        if executable(l:ve_executable)
+            return l:ve_executable
+        endif
+    endif
+
+    if executable(a:cmd_name)
+        return a:cmd_name
+    endif
+
+    return v:null
```
Comment (w0rp) [inline autoload/ale/handlers/python.vim]: You can return an empty string here. I recommend only using `v:null` for JSON.

File: autoload/ale/fixers/yapf.vim [hunk 2]
```diff
@@ -1,13 +1,18 @@
 " Author: w0rp <devw0rp@gmail.com>
 " Description: Fixing Python files with yapf.
 
-function! ale#fixers#yapf#Fix(buffer) abort
+function! ale#fixers#yapf#Fix(buffer, lines) abort
```
Comment (w0rp) [inline autoload/ale/fixers/yapf.vim]: You can remove the `lines` argument here and return `0` instead. You can return `0` to skip any fixer.

Comment (w0rp) [inline autoload/ale/fixers/yapf.vim]: The same applies to the other files.

File: autoload/ale/fixers/yapf.vim [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+    endif
+
     let l:config = ale#path#FindNearestFile(a:buffer, '.style.yapf')
     let l:config_options = !empty(l:config)
     \   ? ' --style ' . ale#Escape(l:config)
     \   : ''
 
     return {
-    \   'command': 'yapf --no-local-style' . l:config_options,
+    \   'command': l:executable . ' --no-local-style' . l:config_options,
```
Comment (w0rp) [inline autoload/ale/fixers/yapf.vim]: Apply `ale#Escape` to `l:executable` right here, and in other files. The path to virtualenv might contain special characters.

**Comment by jlesquembre** [comment]
I did the changes, but for the tests I need some more time, never used vader before.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**